pkg/controller/versions/strategy_v2_0.go

```diff
@@ -404,6 +412,12 @@ func (v *versionStrategyV2_0) Render(ctx context.Context, cr *common.ControllerR
 	return renderings, nil
 }
 
+func validateAppliedSpec(appliedSpec *v2.ControlPlaneSpec) error {
```

Comment:
Would it not be better to try to run this through the regular validation process, i.e. use the same validation logic that's used by the webhook?

Reply:
It can't be the same, as that would mean that the webhook would reject the SMCP in cases where the SMCP becomes valid only after applying the profiles. But I guess we could call the webhook validation logic here and add the additional checks that require the profiles to be applied beforehand. I'll do this.

Reply:
OK, I've now changed this so that full validation is performed just before the charts are rendered.